src/krisi/evaluate/scorecard.py

```diff
@@ -1,29 +1,209 @@
+import logging
+from collections.abc import Iterable
 from dataclasses import dataclass
 from enum import Enum
-from typing import Any, Optional
+from typing import Any, Generic, List, Optional, Tuple, TypeVar, Union
+
+import numpy as np
+
+from krisi.utils.printing import bold, get_term_size, iterative_length
 
 
 class SampleTypes(Enum):
     insample = "insample"
     outsample = "outsample"
 
 
+class MCats(Enum):
+    residual = "Residual Diagnostics"
+    entropy = "Information Entropy"
+    class_err = "Forecast Errors - Classification"
+    reg_err = "Forecast Errors - Regression"
+
+
+T = TypeVar("T", bound=Union[float, int, str, List, Tuple])
+
+
+@dataclass
+class Metric(Generic[T]):
+    name: str
+    category: Optional[MCats] = None
+    metric_result: Optional[T] = None
+    hyperparameters: Optional[Any] = None
```

Comment:
I think Metric can be immutable, so we can remove the setters and the @dataclass will automatically create an init function based on this.

Reply:
ahh okay, we can't, I see!